File: bqlint/parser/token.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-


class Token(object):
    COMMA = 'Comma'
    BRACKET = 'Bracket'
    KEYWORD = 'Keyword'
    OPERATOR = 'Operator'
    COMMENT = 'Comment'
    IDENTIFIER = 'Identifier'
    WHITESPACE = 'Whitespace'
    UNKNOWN = 'Unknown'

    def __init__(self, word, kind=UNKNOWN):
        self.word = word
        self.kind = kind

    def __len__(self):
        return len(self.word)

    def __str__(self):
        return "<{}: '{}'>".format(self.kind, self.word)

    def __repr__(self):
        return "<{}: '{}'>".format(self.kind, self.word)

File: chrome-extension/token.js

/**
 *
 */
class Token {
    static COMMA = 'Comma';
    static BRACKET = 'Bracket';
    static KEYWORD = 'Keyword';
    static OPERATOR = 'Operator';
    static COMMENT = 'Comment';
    static IDENTIFIER = 'Identifier';
    static WHITESPACE = 'Whitespace';
    static UNKNOWN = 'Unknown';

    constructor(word, kind = Token.UNKNOWN) {
        this.word = word;
        this.kind = kind;
    }

    get word() {
        return this._word;
    }

    set word(value) {
        this._word = value;
    }

    get kind() {
        return this._kind;
    }

    set kind(value) {
        if (value.length === 0) {
            value = 'unknown';
        }

        this._kind = value;
    }

    toString() {
        return `<"${this.kind}", "${this.word}">`;
    }
}


/**
 *
 */
class BigQueryToken {
    constructor(word, kind) {
        this.word = word;
        this.kind = kind;
    }

    get word() {
        return this._word;
    }

    set word(value) {
        this._word = value;
    }

    get kind() {
        return this._kind;
    }

    set kind(value) {
        if (value.length === 0) {
            value = 'unknown';
        }

        this._kind = value;
    }

    toString() {
        return `<"${this.kind}", "${this.word}">`;
    }
}
